### PR TITLE
Fixes gatling guns causing horrific lag on spawn

### PR DESCRIPTION
**Pull request by silicons**
?!!?! this might not even fix it but I think it'll atleast make it better?
Anyways these lag HORRIFICALLY when you spawn them in and I have no idea why.

### CODE REVIEW

**Comment by jughu** [comment]
I thought we disabled gatling guns for a reason mekhi 🤔

**Comment by optimumtact** [comment]
This doesn't strike me as being something that would have any effect, did you test it at all or even put any debugging effort into figuring out why they cause bad performance?

**Comment by Grazyn** [comment]
I was there and I recall that there were like 100 energy backpacks on the same tile, this may have something to do with it

**Comment by kevinz000** [comment]
@optimumtact I'm under the impression that using Initialize is better than New() for this stuff?

**Comment by FantasticFwoosh** [comment]
Gatling guns were ungodly powerful against antags (specifically blobs) im not sure they need a reintroduction unless you could actually balance the object.

**Comment by Cyberboss** [comment]
@kevinz000 New and Initialize aren't really that different. They are both called (eventually) when an object is created. If something is creating lag in that proc, this won't fix it.

File: code/modules/projectiles/guns/ballistic/laser_gatling.dm [hunk 1]
```diff
@@ -15,10 +15,10 @@
 	var/overheat_max = 40
 	var/heat_diffusion = 1
 
-/obj/item/weapon/minigunpack/New()
+/obj/item/weapon/minigunpack/Initialize()
```
Comment (Cyberboss) [inline code/modules/projectiles/guns/ballistic/laser_gatling.dm]: @kevinz000 New and Initialize aren't really that different. They are both called (eventually) when an object is created. If something is creating lag in that proc, this won't fix it.

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by optimumtact** [comment]
@kevinz000 we do have better control over how Intialize procs are set and run and they're easier to profile, but it wont' actually just automatically fix the lag.

**Pull request closed without merging**